File: Programmers/BoHwa/5주차(DFS, BFS)/점프와 순간 이동.java
import java.util.*;

public class Solution {
    public int solution(int n) {
        int ans = 0;

        while(n!=0){
            if (n%2==0){
                n=n/2;//2로 나누었을 때의 몫!
            }
            else {
                n--;
                ans++;
            }
        }

        return ans;
    }
}

File: Programmers/jsh/5주차(BFS,DFS)/점프와 순간 이동.py
def solution(n):
    ans = 0 
    '''
    2의제곱수인지 판별(관련 라이브러리가 있는지 찾아보다가 발견)
    비트연산으로 판별 ex)2의제곱수들은 0,10,100,1000 이므로 1작은수와 and연산시 0이됨
    '''
    if not n&(n-1):
        return 1
    while n!=1: #n이 1이될때까지 2로나누거나 1빼기
        if n%2==0:
            n=n//2
        else:
            n=n-1
            ans+=1
    return ans+1
